poc/flp_pine.py

```diff
@@ -18,11 +19,18 @@
 
 
 class PineValid(
-        Valid[
-            list[float],  # Measurement
-            list[float],  # AggResult
-            F,
-        ]):
+    Valid[
+        list[float],  # Measurement
+        list[float],  # AggResult
+        F,
+    ],
+    metaclass=ABCMeta
+):
+    """
```

Comment:
Alternatively we can factor the common codes into some utility class and use "has-a" relationship.

Reply:
What's your goal here?

Reply:
Just want to mention the alternative approach I have considered.

Reply:
I think we can stick with the approach in the base VDAF poc.

Reply:
* I relocated all the evaluation sub-circuits.
* But the encoding routines are shared by both circuits such that both can be used individually.
* Also Most of parameters are shared by both too. One reason is that we initialize them together, with dependencies on each other. Two initialization of these parameters would be a little bit tricky? Lots of the parameters are used by the shared routines as well, e.g. unpack_encoded_measurement.